### DISCUSSION TITLE
Ideas from the demosceners

### DISCUSSION THREAD

**Discussion by nesbox** [nesbox/TIC-80#1286] [Ideas]
There is a group of us demosceners using the TIC80 for sizecoding tiny programs in <256 or <128 byte TIC files.... 
This is a whole subculture doing this for various platforms. This is because TIC80 is more suitable for sizecoding than say a PICO-8 and therefore could be a driver of cool demonstration programs for the TIC80

Something that is really holding us back:

1) loading a TIC file without filled palette chunk will revert to pre 0.8 palette/waveforms, that is quite unusable for direct color access without wasting bytes/chars on color-index remapping... same for waveforms being initialised at random.. a single byte in the TIC file format for legacy <0.8 defaults (=default/0 value) and 0.8 defaults (value=1) would be perfect.. most of the sizecoded small <256byte TIC files only have a (compressed) code chunk +4b chunkheader and nothing else to go with

2) a special demoscene category with finished/competitive demos and intros on tic80.com / surf would be icing on the cake.. right now we have a list at pouet https://www.pouet.net/prodlist.php?platform%5B%5D=TIC-80&page=1  that will grow considerably over the next few months with dedicated TIC80 intro competitions

3) a commandline option for forcing crt filter (F6) upon startup

Point 1 and 3 should be relatively easy to implement

**Comment by nesbox** [comment]
To force CRT filter pls use `--crt` option on startup, it's already implemented in 0.90
https://github.com/nesbox/TIC-80/blob/61b6515c0451cbc025d3c9bb57446aa4d34d52ea/src/studio/studio.c#L2022

**Reply by superogue** [reply]
thanks :D

**Comment by nesbox** [comment]
2. I'm thinking about how to do it better, maybe we could add a new subsection to the `Demo` or maybe add tags or add a new `Demoscene` category

**Reply by superogue** [reply]
I'm assuming more TIC80 demoscene productions will come this year (already a handful in the first week of january).  
Somehow dividing the demos between game/tech demos and demoscene stuff in the form of officially released audiovisual presentations would be a good division. Alternatively: you could consider moving some more game/engine oriented material to the WIP/tech section.

An always updated list of released demoscene TIC-80 productions can be found at: [https://demozoo.org/platforms/92/](url)

**Reply by nesbox** [reply]
A new `Demoscene` section was added, also the `Demos` section renamed to `Tech`
https://tic80.com/play?cat=5

**Comment by nesbox** [comment]
1. It's a bug if you see randomly initialized waveforms and should be fixed. As for restoring the old palette, it's a workaround to not break compatibility with old carts, I'll think about how to fix it.

**Reply by superogue** [reply]
Great to hear the waveforms are on the to-fix list, this helps a lot with sound (a lot of sizecoded prods are missing sound for this reason).

One way to fix the sweetie-pal / db16 pal selection (and future pal selection) in a compatible way that requires as little bytes as possible could be to include a special chunk with no data for default pal selection (not including this chunk or a chunk with 0 data would be reverting to <0.8 compatible palette and for the least amount of bytes (3) have the reserved third chunk byte select the palette and have a 0 length chunk with no data...  or alternatively a 1 byte length chunk with 1 data byte (4 bytes total).

**Reply by nesbox** [reply]
I just added `CHUNK_DEFAULT` chunk to the cart (#1308), so the Sweetie16 palette and default waveforms will be loaded if the chunk exists.
